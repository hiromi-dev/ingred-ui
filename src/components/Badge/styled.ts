import styled from "styled-components";

export type BadgeType = "normal" | "pill";

type Props = {
  color: string;
  backgroundColor: string;
  type: BadgeType;
}

export const Container = styled.span<Props>`
  display: inline-block;
  padding: 0.25em 0.7em;
  border-radius: ${({ type }) => type === "pill" ? "1em" : "4px"};
  background-color: ${({ backgroundColor }) => backgroundColor};
  text-align: center;
  font-size: 0.65em;
  font-weight: bold;
  color: ${({ color }) => color};
  text-decoration: none;
`;
