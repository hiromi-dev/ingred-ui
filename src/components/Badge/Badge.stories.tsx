import * as React from "react";
import styled from "styled-components";
import Badge from ".";
import Spacer from "../Spacer";
import Typography from "../Typography";

const Container = styled.div`
  padding: ${({ theme }) => theme.spacing * 3}px;
  background-color: ${({ theme }) => theme.palette.background.dark};
`;

export default {
  title: "Badge",
  parameters: {
    component: Badge
  }
};

export const Overview = () => {
  return (
    <Container>
      <Typography size="xxxl">
        Example text{" "}
        <Badge type="primary">hoge</Badge>
      </Typography>
      <Spacer py={1} />
      <Typography size="xxl">
        Example text{" "}
        <Badge type="secondary">hoge</Badge>
      </Typography>
      <Spacer py={1} />
      <Typography size="xl">
        Example text{" "}
        <Badge type="warning">hoge</Badge>
      </Typography>
      <Spacer py={1} />
      <Typography size="lg">
        Example text{" "}
        <Badge type="danger">hoge</Badge>
      </Typography>
      <Spacer py={1} />
      <Typography size="md">
        Example text{" "}
        <Badge type="success">hoge</Badge>
      </Typography>
    </Container>
  );
};
