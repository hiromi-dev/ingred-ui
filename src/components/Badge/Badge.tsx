import * as React from "react";
import * as Styled from "./styled";
import { Theme } from "../../themes";
import { useTheme } from "../../themes/useTheme";

export type BadgeType =
  | "primary"
  | "secondary"
  | "success"
  | "warning"
  | "danger"

const getColor = (key: BadgeType, theme: Theme) => {
  switch (key) {
    case "primary":
      return theme.palette.primary.main;
    case "secondary":
      return theme.palette.gray.dark;
    case "success":
      return theme.palette.success.main;
    case "warning":
      return theme.palette.warning.main;
    case "danger":
      return theme.palette.danger.main;
  }
};

export type Props = React.ComponentPropsWithRef<"a"|"span"> & {
  type: BadgeType;
  component?: "span" | "a";
}

const Badge: React.FunctionComponent<Props> = ({
  type,
  component = "span",
  children,
  ...rest
}) => {
  const theme = useTheme();
  return (
    <Styled.Container
      as={component}
      color={theme.palette.text.white}
      backgroundColor={getColor(type, theme)}
      {...rest}
    >
      {children}
    </Styled.Container>
  )
}

export default Badge;
