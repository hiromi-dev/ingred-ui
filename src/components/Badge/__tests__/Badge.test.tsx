import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import Badge from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("Badge component testing", () => {
  afterEach(cleanup);

  test("Badge", () => {
    const { asFragment } = renderWithThemeProvider(
      <Badge type="primary">text</Badge>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
