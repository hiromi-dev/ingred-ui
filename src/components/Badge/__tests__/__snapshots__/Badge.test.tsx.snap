// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component testing Badge normal 1`] = `
<DocumentFragment>
  <span
    class="sc-AxjAm gbGUAi"
    color="#FFFFFF"
    type="normal"
  >
    text
  </span>
</DocumentFragment>
`;

exports[`Badge component testing Badge pill 1`] = `
<DocumentFragment>
  <span
    class="sc-AxjAm jSrxBj"
    color="#FFFFFF"
    type="pill"
  >
    text
  </span>
</DocumentFragment>
`;
